tests/test_web_functional.py

```diff
@@ -1090,19 +1100,32 @@ async def handler(request: web.Request) -> web.Response:
     resp.release()
 
 
-@pytest.mark.parametrize(
-    "compressor,encoding",
-    [
-        (zlib.compressobj(wbits=16 + zlib.MAX_WBITS), "gzip"),
-        (zlib.compressobj(wbits=zlib.MAX_WBITS), "deflate"),
-        # Actually, wrong compression format, but
-        # should be supported for some legacy cases.
-        (zlib.compressobj(wbits=-zlib.MAX_WBITS), "deflate"),
-    ],
-)
+@pytest.fixture(params=["gzip", "deflate", "deflate-raw"])
+def compressor_case(
+    request: pytest.FixtureRequest, parametrize_zlib_backend: None
+) -> Generator[Tuple[ZLibCompressorObjProtocol, str], None, None]:
+    encoding: str = request.param
+    max_wbits: int = ZLibBackend.MAX_WBITS
+
+    if encoding == "gzip":
+        wbits = 16 + max_wbits
+    elif encoding == "deflate":
+        wbits = max_wbits
+    elif encoding == "deflate-raw":
+        wbits = -max_wbits
+    else:
+        raise ValueError(f"Unknown encoding: {encoding}")
+
```

Comment:
Could be a const `dict` of values based on the encoding, so it would raise `KeyError` if its missing and you wouldn't have have the branching in here